### PR TITLE
Version check cleanup

### PR DESCRIPTION
**Pull request by compnerd**
Replace this paragraph with a description of your changes and rationale. Provide links to external references/discussions if appropriate.

Resolves [SR-NNNN](https://bugs.swift.org/browse/SR-NNNN).

### CODE REVIEW

**Comment by compnerd** [comment]
@swift-ci please test and merge

File: CMakeLists.txt [hunk 1]
```diff
@@ -322,13 +322,14 @@ endfunction()
...
-    find_version("ninja" "--version" TRUE)
-    find_version(${CMAKE_MAKE_PROGRAM} "--version" FALSE)
-  elseif(CMAKE_MAKE_PROGRAM MATCHES "make")
-    find_version("make" "--version" TRUE)
+  message(STATUS "Finding version for: ${CMAKE_COMMAND}")
+  message(STATUS "Found version: ${CMAKE_VERSION}")
+  message(STATUS "")
+
+  if(CMAKE_MAKE_PROGRAM STREQUAL "ninja" OR CMAKE_MAKE_PROGRAM STREQUAL "make")
+    get_filename_component(CMAKE_MAKE_PROGRAM_BN "${CMAKE_MAKE_PROGRAM}" NAME_WE)
```
Comment (rintaro) [inline CMakeLists.txt]: If the condition is `true`, `CMAKE_MAKE_PROGRAM_BN` is always the same as `CMAKE_MAKE_PROGRAM`, isn't it?

Comment (compnerd) [inline CMakeLists.txt]: Oh, that is a good point.  This can be made simpler by just hoisting the get_filename_component.

**Comment by compnerd** [comment]
@swift-ci please test and merge

**Comment by compnerd** [comment]
@swift-ci please smoke test OS X platform

**Pull request merged**